spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -87,18 +87,27 @@ function restorePath(portIdentifier: Identifier, channelIdentifier: Identifier):
 }
 ```
 
-#### UpgradeError Path
+#### CurrentUpgradeSequencePath
 
-The upgrade error path is a public path that can signal an error of the upgrade to the counterparty. It does not store anything in the successful case, but it will store a sentinel abort value in the case that a chain does not accept the proposed upgrade.
+The current upgrade sequence path is a public path that stores the sequence of the current upgrade attempt. The sequence will increment with each attempted upgrade on the given channel. The sequence will be used to ensure that different error receipts referring to different upgrade attempts do not interfere with each other.
 
 ```typescript
-function errorPath(portIdentifier: Identifier, channelIdentifier: Identifier): Path {
-    return "channelUpgrade/ports/{portIdentifier}/channels/{channelIdentifier}/upgradeError"
+function currentSequencePath(portIdentifier: Identifier, channelIdentifier: Identifier) Path {
+    return "channelUpgrade/ports/{portIdentifier}/channelIdentifier/{channelIdentifier}/currentUpgradeSequence"
+}
+```
+
+#### UpgradeError Path
+
+The upgrade error path is a public path that can signal an error of the upgrade to the counterparty for the given upgrade attempt. It does not store anything in the successful case, but it will store a sentinel abort value in the case that a chain does not accept the proposed upgrade. The error path will contain the portIdentifier, channelIdentifier, **and** the upgrade sequence so that the error receipts for different upgrade attempts do not interfere with each other.
```

Comment:
Wouldn't it be sufficient to write the upgrade sequence to the same path that contains only the portIdentifier and channelIdentifier, i.e., `channelUpgrade/ports/{portIdentifier}/channels/{channelIdentifier}/upgradeError`?

Reply:
In order to do that, we would need to create a new data structure for error receipt so that we can parse out the sequence from the error receipt. With this solution we can keep the error receipt to be arbitrary bytes

The other downside is that error receipts will get overwritten on subsequent upgrades, rather than having some store of past attempts that can then be pruned if desired.

For these reasons, since there's only one piece of data we want to communicate in protocol, it made more sense to encode it in the key

Reply:
I ended up having to do as you suggested, since in timeout and confirm we want to prove there is no error receipt stored on the counterparty (or it is only at lower sequence) and this would be difficult to do if sequence is in the key